homeassistant/components/husqvarna_automower/lawn_mower.py

```diff
@@ -0,0 +1,124 @@
+) -> None:
+    """Set up vacuum platform."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        HusqvarnaAutomowerEntity(coordinator, idx)
+        for idx, ent in enumerate(coordinator.session.data["data"])
+    )
+    entity_platform.current_platform.get()
+
+
+class HusqvarnaAutomowerEntity(LawnMowerEntity, AutomowerEntity):
+    """Defining each mower Entity."""
+
+    _attr_name: str | None = None
+    _attr_supported_features = SUPPORT_STATE_SERVICES
+
+    def __init__(self, session, idx) -> None:
+        """Set up HusqvarnaAutomowerEntity."""
+        super().__init__(session, idx)
+        self._attr_unique_id = self.coordinator.session.data["data"][self.idx]["id"]
+
+    @property
+    def available(self) -> bool:
+        """Return True if the device is available."""
+        available = self.get_mower_attributes()["metadata"]["connected"]
+        return available
+
+    @property
+    def activity(self) -> LawnMowerActivity:
+        """Return the state of the mower."""
+        mower_attributes = AutomowerEntity.get_mower_attributes(self)
+        if mower_attributes["mower"]["state"] in [
+            "PAUSED",
+            "WAIT_UPDATING",
+            "WAIT_POWER_UP",
+        ]:
+            activity = LawnMowerActivity.PAUSED
+        if mower_attributes["mower"]["activity"] in ["MOWING", "LEAVING", "GOING_HOME"]:
+            activity = LawnMowerActivity.MOWING
+        if (mower_attributes["mower"]["state"] == "RESTRICTED") or (
+            mower_attributes["mower"]["activity"] in ["PARKED_IN_CS", "CHARGING"]
+        ):
+            activity = LawnMowerActivity.DOCKED
+        if (
+            mower_attributes["mower"]["state"]
+            in [
+                "FATAL_ERROR",
+                "ERROR",
+                "ERROR_AT_POWER_UP",
+                "NOT_APPLICABLE",
+                "UNKNOWN",
+                "STOPPED",
+                "OFF",
+            ]
+        ) or mower_attributes["mower"]["activity"] in ERROR_ACTIVITIES:
+            activity = LawnMowerActivity.ERROR
+        return LawnMowerActivity(activity)
+
+    @property
+    def extra_state_attributes(self) -> dict:
+        """Return the specific state attributes of this mower."""
+        mower_attributes = AutomowerEntity.get_mower_attributes(self)
+        action = mower_attributes["planner"]["override"]["action"]
+        action = action.lower() if action is not None else action
+        return {
+            "action": action,
+        }
+
+    async def async_start_mowing(self) -> None:
+        """Resume schedule."""
+        command_type = "actions"
+        payload = '{"data": {"type": "ResumeSchedule"}}'
+        try:
+            await self.coordinator.session.action(self.mower_id, payload, command_type)
+        except ClientResponseError as exception:
+            _LOGGER.error("Command couldn't be sent to the command que: %s", exception)
```

Comment:
I think you can extract this functionality into a separate function where you only have to pass in the type, so for example
`execute_action("ResumeSchedule")`

Reply:
The library client should expose easy to use methods. The library user shouldn't need to know to pass a `" ResumeSchedule"` string. Create different methods for different actions, resume_schedule`, `start_mowing` etc.

Reply:
I liked it, when the library was dumb, at could just pass any commands. This made it very flexible. But I changed the library. -> In my opinion it's now resolved. Let me know, if not.